### PR TITLE
add nz-affix & nz-anchor & nz-back-top components.

### PR DESCRIPTION
**Pull request by cipchk**

### CODE REVIEW

**Comment by LinBoLen** [comment]
awesome

File: src/components/affix/nz-affix.component.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  EventEmitter,
+  Output,
+  Renderer2,
+  OnDestroy,
+  ViewChild,
+  ElementRef,
+  HostBinding
+} from '@angular/core';
+import { Subscription } from 'rxjs/Subscription';
+import { Observable } from 'rxjs/Rx';
```
Comment (trotyl) [inline src/components/affix/nz-affix.component.ts]: Should only import specific type as suggested in https://github.com/NG-ZORRO/ng-zorro-antd/pull/29

File: src/components/anchor/nz-anchor.component.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  Output,
+  Renderer2,
+  OnDestroy,
+  ViewChild,
+  ElementRef,
+  Inject
+} from '@angular/core';
+import { DOCUMENT } from '@angular/platform-browser';
+import { Subscription } from 'rxjs/Subscription';
+import { Observable } from 'rxjs/Rx';
```
Comment (trotyl) [inline src/components/anchor/nz-anchor.component.ts]: Should only import specific type as suggested in https://github.com/NG-ZORRO/ng-zorro-antd/pull/29

File: src/components/back-top/nz-back-top.component.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+} from '@angular/core';
+
+import {
+  trigger,
+  style,
+  transition,
+  animate
+} from '@angular/animations';
+import { Subscription } from 'rxjs/Subscription';
+import { Observable } from 'rxjs/Rx';
```
Comment (trotyl) [inline src/components/back-top/nz-back-top.component.ts]: Should only import specific type as suggested in https://github.com/NG-ZORRO/ng-zorro-antd/pull/29

**Comment by trotyl** [comment]
Would it better to merge into one commit? It would be cleaner in log and easier to revert.

**Comment by cipchk** [comment]
是否还有其它情况呢？目前 `ng test` 实在很无感，待这个FIX解决后补上。

**Comment by vthinkxie** [comment]
@cipchk please follow this contributing guide format, thanks very much.
https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit

**Comment by cipchk** [comment]
之前没有一些原因的关系倒置没有区分，把三个组件一次性提交了。我先关掉，待我晚上重新拆分成独立后再提交。

**Pull request closed without merging**